File: guides/java/kora-java-guide-dependency-injection-introduction-app/src/test/java/ru/tinkoff/kora/guide/dependencyinjection/NotificationServiceTest.java
package ru.tinkoff.kora.guide.dependencyinjection;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.List;
import org.junit.jupiter.api.Test;
import ru.tinkoff.kora.test.extension.junit5.KoraAppTest;
import ru.tinkoff.kora.test.extension.junit5.TestComponent;

@KoraAppTest(Application.class)
class NotificationServiceTest {

    @TestComponent
    private NotificationService notificationService;

    @Test
    void broadcast_ShouldUseAllNotifiers() {
        List<String> result = notificationService.broadcast("Hello");

        assertEquals(2, result.size());
        assertTrue(result.contains("EMAIL: [app] Hello"));
        assertTrue(result.contains("SMS: [app] Hello"));
    }

    @Test
    void notifyEmailOnly_ShouldResolveTaggedNotifier() {
        String result = notificationService.notifyEmailOnly("Ping");

        assertEquals("EMAIL: [app] Ping", result);
    }

    @Test
    void optionalAuditDependency_ShouldBeHandledGracefully() {
        assertFalse(notificationService.isAuditEnabled());
    }
}

File: guides/kotlin/kora-kotlin-guide-dependency-injection-introduction-app/src/test/kotlin/ru/tinkoff/kora/guide/dependencyinjection/NotificationServiceTest.kt
package ru.tinkoff.kora.guide.dependencyinjection

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import ru.tinkoff.kora.test.extension.junit5.KoraAppTest
import ru.tinkoff.kora.test.extension.junit5.TestComponent

@KoraAppTest(Application::class)
class NotificationServiceTest {
    @TestComponent
    lateinit var notificationService: NotificationService

    @Test
    fun broadcastUsesAllTaggedAndUntaggedNotifiers() {
        val result = notificationService.broadcast("hello")
        assertEquals(2, result.size)
        assertTrue(result.contains("EMAIL: [app] hello"))
        assertTrue(result.contains("SMS: [app] hello"))
    }

    @Test
    fun emailOnlyUsesTaggedComponent() {
        assertEquals("EMAIL: [app] hello", notificationService.notifyEmailOnly("hello"))
    }

    @Test
    fun optionalAuditSinkIsAbsent() {
        assertFalse(notificationService.isAuditEnabled())
    }
}
